### PR TITLE
removed unsafe property access from ReferenceCounter

### PR DESCRIPTION
**Pull request by sebastianbaginski**
Fixes bug mentioned in https://github.com/magic-lang/ooc-kean/pull/1811
@marcusnaslund

### CODE REVIEW

**Comment by sebastianbaginski** [comment]
Fun fact, it does not generate the `isSafe` data member now:
```ooc
struct _ReferenceCounter__ReferenceCounter {
    struct _types__Object __super__;
    types__Object* _target;
    Numbers__Int _count;
    types__Bool _kill;
    Mutex__Mutex* _lock;
};
```

File: test/base/ReferenceCounterTest.ooc [hunk 1]
```diff
@@ -43,6 +43,11 @@ ReferenceCounterTest: class extends Fixture {
 		object := TestObject new(isAlive&)
 		counter := ReferenceCounter new(object)
 		counter isSafe = true
+		expect(counter isSafe, is equal to(true))
+		counter isSafe = false
+		expect(counter isSafe, is equal to(false))
+		counter isSafe = true
+		expect(counter isSafe, is equal to(true))
```
Comment (marcusnaslund) [inline test/base/ReferenceCounterTest.ooc]: use `is True` and `is False` instead

**Comment by marcusnaslund** [comment]
Very good. Have you gone through the other setters, or should I?

**Comment by sebastianbaginski** [comment]
> Have you gone through the other setters, or should I?

Nope, feel free to fix the other sources if you find some time.

File: test/base/ReferenceCounterTest.ooc (hunk 1 above)
Comment (sebastianbaginski) [inline test/base/ReferenceCounterTest.ooc]: Updated.

**Pull request merged**